apps/dashboard/lib/auth/server.ts

```diff
@@ -0,0 +1,47 @@
+import { env } from "@/lib/env";
+
+import type { BaseAuthProvider } from "./base-provider";
+import { LocalAuthProvider } from "./local";
+import { WorkOSAuthProvider } from "./workos";
+
+class AuthProvider {
+  private static instance: BaseAuthProvider | null = null;
+  private static initialized = false;
+
+  private static initialize(): void {
+    if (this.initialized) {
+      return;
+    }
+
+    const authProvider = env().AUTH_PROVIDER;
```

Comment:
not a huge deal, but calling env() 3 times will parse and validate all variables 3 times

it'd be better if we only call it once